rasterio/_base.pyx

```diff
@@ -25,16 +24,7 @@ from rasterio.enums import (
 from rasterio.vfs import parse_path, vsi_path
 
 
-log = logging.getLogger('rasterio')
-if 'all' in sys.warnoptions:
-    # show messages in console with: python -W all
-    logging.basicConfig()
-else:
-    # no handler messages shown
-    class NullHandler(logging.Handler):
-        def emit(self, record):
-            pass
-    log.addHandler(NullHandler())
```

Comment:
@sgillies I have seen others mention that this should not be done but have never encountered a reason.  Without taking too much of your time, could you briefly describe why `logging.basicConfig()` is bad?